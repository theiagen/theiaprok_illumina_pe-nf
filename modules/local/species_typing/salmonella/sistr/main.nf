process SISTR {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "us-docker.pkg.dev/general-theiagen/staphb/sistr:1.1.3"

    input:
    tuple val(meta), path(assembly)

    output:
    tuple val(meta), path("*.tsv")             , emit: sistr_result
    tuple val(meta), path("*-allele.json")     , emit: sistr_allele_json
    tuple val(meta), path("*-allele.fasta")    , emit: sistr_allele_fasta
    tuple val(meta), path("*-cgmlst.csv")      , emit: sistr_cgmlst
    tuple val(meta), path("*_predicted_serotype.txt")     , emit: sistr_predicted_serotype
    tuple val(meta), path("*_serogroup.txt")              , emit: sistr_serogroup
    tuple val(meta), path("*_h1_antigens.txt")            , emit: sistr_h1_antigens
    tuple val(meta), path("*_h2_antigens.txt")            , emit: sistr_h2_antigens
    tuple val(meta), path("*_o_antigens.txt")             , emit: sistr_o_antigens
    tuple val(meta), path("*_serotype_cgmlst.txt")        , emit: sistr_serotype_cgmlst
    tuple val(meta), path("*_antigenic_formula.txt")      , emit: sistr_antigenic_formula
    path "versions.yml"                        , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def use_full_cgmlst_db = params.sistr_use_full_cgmlst_db ? '--use-full-cgmlst-db' : ''
    
    """
    sistr \\
        --qc \\
        ${use_full_cgmlst_db} \\
        --threads ${task.cpus} \\
        --alleles-output ${prefix}-allele.json \\
        --novel-alleles ${prefix}-allele.fasta \\
        --cgmlst-profiles ${prefix}-cgmlst.csv \\
        --output-prediction ${prefix} \\
        --output-format tab \\
        ${args} \\
        ${assembly}
    
    # Rename to .tsv suffix
    mv ${prefix}.tab ${prefix}.tsv

    # Parse sistr TSV
    cut -f 16 ${prefix}.tsv | tail -n 1 > ${prefix}_predicted_serotype.txt
    cut -f 15 ${prefix}.tsv | tail -n 1 > ${prefix}_serogroup.txt
    cut -f 10 ${prefix}.tsv | tail -n 1 > ${prefix}_h1_antigens.txt
    cut -f 11 ${prefix}.tsv | tail -n 1 > ${prefix}_h2_antigens.txt
    cut -f 12 ${prefix}.tsv | tail -n 1 > ${prefix}_o_antigens.txt
    cut -f 18 ${prefix}.tsv | tail -n 1 > ${prefix}_serotype_cgmlst.txt
    cut -f 1 ${prefix}.tsv | tail -n 1 > ${prefix}_antigenic_formula.txt
    
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        sistr: \$(sistr --version 2>&1 | sed 's/^.*sistr_cmd //; s/ .*\$//')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.tsv
    touch ${prefix}-allele.json
    touch ${prefix}-allele.fasta
    touch ${prefix}-cgmlst.csv

    touch ${prefix}_predicted_serotype.txt
    touch ${prefix}_serogroup.txt
    touch ${prefix}_h1_antigens.txt
    touch ${prefix}_h2_antigens.txt
    touch ${prefix}_o_antigens.txt
    touch ${prefix}_serotype_cgmlst.txt
    touch ${prefix}_antigenic_formula.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        sistr: \$(sistr --version 2>&1 | sed 's/^.*sistr_cmd //; s/ .*\$//')
    END_VERSIONS
    """
}