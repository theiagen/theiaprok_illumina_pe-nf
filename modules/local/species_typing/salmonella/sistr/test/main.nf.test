nextflow_process {
    name "Test Process SISTR"
    script "../main.nf"
    process "SISTR"

    test("SISTR - test") {
        setup {
            run("GUNZIP") {
                script "../../../../../nf-core/gunzip/main.nf"
                process {
                    """
                    input[0] = [
                        [id:'test_sample'],
                        [file('https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/945/GCF_000006945.2_ASM694v2/GCF_000006945.2_ASM694v2_genomic.fna.gz', checkIfExists: true)]
                    ]
                    """
                }
            }
        }
        when {
            process {
                """
                input[0] = GUNZIP.out.gunzip
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot().match() }
            )
        }
    }

    test("SISTR - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test_sample' ],
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/escherichia_coli/genome/genome.fa.gz', checkIfExists: true)
                ])
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}