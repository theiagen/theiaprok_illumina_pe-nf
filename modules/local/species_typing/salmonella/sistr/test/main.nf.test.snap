{
    "SISTR - test": {
        "content": [
            
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-06-24T21:40:30.984779695"
    },
    "SISTR - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample-allele.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "10": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample_antigenic_formula.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "11": [
                    "versions.yml:md5,9cd2b041e26756435a56098355b5a2ca"
                ],
                "2": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample-allele.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample-cgmlst.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample_predicted_serotype.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample_serogroup.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample_h1_antigens.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample_h2_antigens.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "8": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample_o_antigens.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "9": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample_serotype_cgmlst.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "sistr_allele_fasta": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample-allele.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "sistr_allele_json": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample-allele.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "sistr_antigenic_formula": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample_antigenic_formula.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "sistr_cgmlst": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample-cgmlst.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "sistr_h1_antigens": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample_h1_antigens.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "sistr_h2_antigens": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample_h2_antigens.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "sistr_o_antigens": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample_o_antigens.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "sistr_predicted_serotype": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample_predicted_serotype.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "sistr_result": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "sistr_serogroup": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample_serogroup.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "sistr_serotype_cgmlst": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample_serotype_cgmlst.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,9cd2b041e26756435a56098355b5a2ca"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-06-24T21:40:43.740233188"
    }
}